docs/contributing.rst

```diff
@@ -1,20 +1,52 @@
 Contributing to py-evm
 ----------------------
 
+First we need to clone the Py-EVM repository. Py-EVM depends on a submodule of the common tests across all clients, so we need to clone the repo with the ``--recursive`` flag. Example:
+
+.. code:: sh
+
+    $ git clone --recursive https://github.com/ethereum/py-evm.git
+
+
+Py-EVM requires Python 3. Often, the best way to guarantee a clean Python 3 environment is with `virtualenv <https://virtualenv.pypa.io/en/stable/>`_, like:
+
+.. code:: sh
+
+    # once:
+    $ virtualenv -p python3 venv
+
+    # each session:
+    $ . venv/bin/activate
+
+After we have activated our virtual environment, installing all dependencies that are needed to run, develop and test all code in this repository is as easy as:
+
 .. code:: sh
 
-    pip install -e . -r requirements-dev.txt
+    pip install -e .[dev]
+
 
 Running the tests
 ~~~~~~~~~~~~~~~~~
 
-You can run the tests with:
+A great way to explore the code base is to run the tests.
+
+We can run all tests with:
```

Comment:
Probably worth an admonition here that naively running the full test suite is likely to take multiple hours to run.

Reply:
Added:

>However, running the entire test suite does take a very long time so often we just want to run a subset instead, like: